cmd/traffic/cmd/manager/internal/state/state.go

```diff
@@ -548,9 +551,26 @@ func (s *State) AddIntercept(sessionID, clusterID, apiKey string, client *rpc.Cl
 		return nil, status.Errorf(codes.AlreadyExists, "Intercept named %q already exists", spec.Name)
 	}
 
+	sess := s.sessions[sessionID].(*clientSessionState)
```

Comment:
This should be checked, even if it's not likely to fail. The `sess` can be obtained right at the beginning of the function and return `status.Errorf(codes.NotFound, "session %q not found", id)` if the session cannot be found and casted.